admin/services/data-access/group.data.service.js

```diff
@@ -20,17 +24,99 @@ groupDataService.getGroups = async function (query) {
     .exec()
 }
 
+/**
+ * Get active groups (non-soft-deleted).
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlFindGroups = async (schoolId) => {
+  const sql = `
+  SELECT g.id, g.name, COUNT(pg.pupil_id) as pupilCount 
+  FROM ${sqlService.adminSchema}.[group] g
+  LEFT OUTER JOIN ${sqlService.adminSchema}.pupilGroup pg 
+  ON g.id = pg.group_id
+  WHERE g.isDeleted=0
+  AND g.school_id=@schoolId
+  GROUP BY g.id, g.name
+  ORDER BY name ASC`
+  const params = [
+    {
+      name: 'schoolId',
+      value: schoolId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
 /**
  * Get group document by _id.
+ * @deprecated use sqlFindGroup
  * @param query
  * @returns {Promise<*>}
  */
 groupDataService.getGroup = async function (query) {
   return Group.findOne(query).lean().exec()
 }
 
+/**
+ * Get group by id.
+ * @param groupId
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlFindGroup = async (groupId, schoolId) => {
+  const sql = `SELECT id, [name] 
+    FROM ${sqlService.adminSchema}.[group]
+    WHERE id=@groupId AND school_id=@schoolId`
+
+  const params = [
+    {
+      name: 'groupId',
+      value: groupId,
+      type: TYPES.Int
+    },
+    {
+      name: 'schoolId',
+      value: schoolId,
+      type: TYPES.Int
+    }
+  ]
+
+  const result = await sqlService.query(sql, params)
+  return R.head(result)
+}
+
+/**
+ * Get group by name.
+ * @param groupName
+ * @returns {Promise<void>}
+ */
+groupDataService.sqlFindGroupByName = async (groupName, schoolId) => {
```

Comment:
Suggest renaming this to `sqlFindOneByName` as per our usual naming conventions.

school_id doesn't seem to appear in the diff for the CREATE TABLE [group].  If it was in there this query would be safer if there was an index on (school_id, name)

Reply:
we cant have an index on school_id & name due to incompatible data types :/

Reply:
regarding index -this will be added now we have reduced col length on name